srv/salt/_modules/dg.py

```diff
@@ -961,7 +961,8 @@ def deploy(**kwargs):
     # do not run this when there is still ceph:storage in the pillar
     # this indicates that we are in a post-upgrade scenario and
     # the drive assignment was not ported to drive-groups yet.
-    if __pillar__.get('ceph', {}).get('storage'):
+    if __pillar__.get('ceph', {}).get('storage') and not kwargs.get(
```

Comment:
shouldn't you write condition like this: `not kwargs.get('bypass_pillar', False) and ('storage' in __pillar__.get('ceph', {}))`? This way you will only check the pillar if `bypass_pillar is False` and will avoid a `KeyError` exception when 'storage' is not a key in the `__pillar__['ceph']`

Reply:
right, that's better.